modules/renter/contractor/allowance.go

```diff
@@ -147,12 +148,10 @@ func (c *Contractor) SetAllowance(a modules.Allowance) error {
 	c.mu.Lock()
 	c.allowance = a
 	c.contracts = newContracts
-	// update metrics
-	var spending types.Currency
-	for _, contract := range c.contracts {
-		spending = spending.Add(contract.RenterFunds())
+	// if the currentPeriod was previously unset, set it now
+	if c.currentPeriod == 0 {
+		c.currentPeriod = periodStart
```

Comment:
The period should only reset if the user is expecting it to reset, which I don't think is directly related to how many contracts are being replaced. I'm trying to imagine what a sane outcome would be if I extended the period. Am I expecting to enter a new period, or am I expecting the current period to roll over?

I'm sort of thinking that the user would not be expecting to enter a new period.

And same if the period was shortened. If you bring it from 30 days down to 7, you probably want it to start from the previous start, and then walk forward by 7 until the current period is hit. Or maybe you do want it to reset completely?

Not sure, but I think it's okay to punt it for now.